attest panic on MacOS

**What happened**:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x101f13fbc]

goroutine 1 [running]:
github.com/anchore/syft/cmd/syft/cli/attest.formatPredicateType(...)
	/Users/runner/work/syft/syft/cmd/syft/cli/attest/attest.go:323
github.com/anchore/syft/cmd/syft/cli/attest.Run({0x1029367a8, _}, _, {0x0, {0x0, 0x0}, {0x16f7b7ac2, 0x19}, {0x101f8a33d, 0x1b}, ...}, ...)
	/Users/runner/work/syft/syft/cmd/syft/cli/attest/attest.go:75 +0x9c
github.com/anchore/syft/cmd/syft/cli.Attest.func2(0x14001219680?, {0x1400122fc20?, 0x1?, 0x6?})
	/Users/runner/work/syft/syft/cmd/syft/cli/attest.go:70 +0x130
github.com/spf13/cobra.(*Command).execute(0x14001219680, {0x1400122fbc0, 0x6, 0x6})
	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x4d0
github.com/spf13/cobra.(*Command).ExecuteC(0x14001246000)
	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x354
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
	/Users/runner/work/syft/syft/cmd/syft/main.go:15 +0x60
```

**What you expected to happen**:
SBOM produced and signed as attestation stored locally 
**How to reproduce it (as minimally and precisely as possible)**:
`syft attest --key release.key --no-upload -o sdpx-json myregistry.net/my-api:latest`
**Anything else we need to know?**:
release.key was generated by cosign generate-key, although I don't think it matters
**Environment**:
- Output of `syft version`:
 ```
syft version
Application:        syft
Version:            0.56.0
JsonSchemaVersion:  3.3.2
BuildDate:          2022-09-12T16:36:53Z
GitCommit:          c5dca001e267d2a91ff82e53ca72535ceef6af02
GitDescription:     v0.56.0
Platform:           darwin/arm64
GoVersion:          go1.18.5
Compiler:           gc
```

syft is installed with brew
- OS (e.g: `cat /etc/os-release` or similar):
```
 sw_vers 
ProductName:	macOS
ProductVersion:	12.4
BuildVersion:	21F79
```

--- a/cmd/syft/cli/attest/attest.go
+++ b/cmd/syft/cli/attest/attest.go
@@ -73,6 +73,11 @@ func Run(ctx context.Context, app *config.Application, ko sigopts.KeyOpts, args 
 	output := parseAttestationOutput(app.Outputs)
 
 	format := syft.FormatByName(output)
+
+	// user typo or unknown outputs provided
+	if format == nil {
+		format = syft.FormatByID(spdx22json.ID) // default attestation format
+	}
 	predicateType := formatPredicateType(format)
 	if predicateType == "" {
 		return fmt.Errorf(

--- a/cmd/syft/cli/attest/attest.go
+++ b/cmd/syft/cli/attest/attest.go
@@ -76,7 +76,7 @@ func Run(ctx context.Context, app *config.Application, ko sigopts.KeyOpts, args 
 
 	// user typo or unknown outputs provided
 	if format == nil {
-		format = syft.FormatByID(spdx22json.ID) // default attestation format
+		format = syft.FormatByID(syftjson.ID) // default attestation format
 	}
 	predicateType := formatPredicateType(format)
 	if predicateType == "" {
@@ -118,7 +118,7 @@ func Run(ctx context.Context, app *config.Application, ko sigopts.KeyOpts, args 
 
 func parseAttestationOutput(outputs []string) (format string) {
 	if len(outputs) == 0 {
-		outputs = append(outputs, string(spdx22json.ID))
+		outputs = append(outputs, string(syftjson.ID))
 	}
 
 	return outputs[0]

--- a/cmd/syft/cli/attest/attest.go
+++ b/cmd/syft/cli/attest/attest.go
@@ -70,7 +70,9 @@ func Run(ctx context.Context, app *config.Application, ko sigopts.KeyOpts, args 
 		return err
 	}
 
-	format := syft.FormatByName(app.Outputs[0])
+	output := parseAttestationOutput(app.Outputs)
+
+	format := syft.FormatByName(output)
 	predicateType := formatPredicateType(format)
 	if predicateType == "" {
 		return fmt.Errorf(
@@ -107,6 +109,14 @@ func Run(ctx context.Context, app *config.Application, ko sigopts.KeyOpts, args 
 		stereoscope.Cleanup,
 		ui.Select(options.IsVerbose(app), app.Quiet)...,
 	)
+}
+
+func parseAttestationOutput(outputs []string) (format string) {
+	if len(outputs) == 0 {
+		outputs = append(outputs, string(spdx22json.ID))
+	}
+
+	return outputs[0]
 }
 
 func parseImageSource(userInput string, app *config.Application) (s *source.Input, err error) {

--- a/test/cli/attest_cmd_test.go
+++ b/test/cli/attest_cmd_test.go
@@ -40,6 +40,14 @@ func TestAttestCmd(t *testing.T) {
 			},
 			pw: "",
 		},
+		{
+			name: "can encode syft.json as the predicate given a user format typo",
+			args: []string{"attest", "-o", "spdx-jsonx", "--key", "cosign.key", img},
+			assertions: []traitAssertion{
+				assertSuccessfulReturnCode,
+			},
+			pw: "",
+		},
 	}
 
 	for _, test := range tests {

